test/hotspot/jtreg/serviceability/jvmti/vthread/ThreadListStackTracesTest/ThreadListStackTracesTest.java

```diff
@@ -44,6 +44,13 @@ static void sleep(long millis) {
         }
     }
 
+    public void ensureReadyAndWaiting(Thread vt, Thread.State expState, ReentrantLock rlock) {
+        // wait while the thread is not ready or thread state is unexpected
+        while (!threadReady || (vt.getState() != expState) || !rlock.hasQueuedThreads()) {
```

Comment:
Small nit. 
The comment is incomplete. And code duplication is unneded. 

You can make  
```

task.ensureReady(vt, expState);
// Ensure that thread waiting on the rlock
while (!rlock.hasQueuedThreads()) {   
  sleep(1);
}
```
to reduce code duplication. 

Really,
```
task.ensureReady(vt, expState);
```
Shouldn't be needed. We can't have lock.hasQueuedThreads() == true until
we update `threadReady` and hit  `vt.getState()`. 
But I think it is fine to check all conditions.